### PR TITLE
Test for Homework Service

### LINKED ISSUE #383: Add tests for HomeworkService
Currently, there are no unit tests for `HomeworkService`. We need to implement unit tests for this service and cover it's public methods.

### PR DESCRIPTION
**Pull request by QuestroL**
add 4 tests for CreateHomeworkAsync method
add 3 tests for UpdateHomeworkAsync method
add 2 tests for GetHomeworkByLessonIdAsync method
add  2 tests for GetHomeworkByIdAsync method

### CODE REVIEW

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+                AttachmentIds = new List<long> { 1, 2 }
+            };
+
+            _lessonRepositoryMock.Setup(l => l.IsEntityExistAsync(1)).ReturnsAsync(false);
+            //_homeworkRepositoryMock.Setup(x => x.Add(It.IsAny<Homework>()))
+            //        .Callback<Homework>(x => new Homework { LessonId = homeworkRequestDto.LessonId });
+            _attachmentRepositoryMock.Setup(x => x.IsEntityExistAsync(It.IsAny<long>())).ReturnsAsync(false);
+
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            // _unitOfWorkMock.Setup(x => x.HomeworkRepository).Returns(_homeworkRepositoryMock.Object);
```
Comment (Hlib-mats) [inline CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs]: Remove this commented code please, if you don't need it

File: CharlieBackend.Api.UnitTes/StudentServiceTests.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public async Task CreateStudentAsync_ExistingAccountIdNotAssignedStudent_ShouldR
                .ReturnsAsync(successExistingAccount);
 
             InitializeForCreateStudentAsync();
-
+
```
Comment (Hlib-mats) [inline CharlieBackend.Api.UnitTes/StudentServiceTests.cs:54]: Remove this space

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z")
+            };
+
+            Homework homework = new Homework
+            {
+                Id = 1,
+                LessonId = 1,
+                DueDate = DateTime.Parse("2021-11-18T15:00:00.384Z"),
+
```
Comment (Hlib-mats) [inline CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs:185]: remove this empty line please

Comment (Hlib-mats) [reply CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs:185]: Also check this file for empty lines

**Review by Hlib-mats — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+            _homeworkRepositoryMock.Setup(x => x.GetByIdAsync(It.IsAny<long>())).ReturnsAsync(homework);
+
+            _unitOfWorkMock.Setup(x => x.HomeworkRepository).Returns(_homeworkRepositoryMock.Object);
+            _unitOfWorkMock.Setup(x => x.LessonRepository).Returns(_lessonRepositoryMock.Object);
+            //Act
+
+            var result = await _homeworkService.UpdateHomeworkAsync(1, homeworkRequestDto);
+
+            //Assert
+            result.Data.Should().BeEquivalentTo(new HomeworkDto
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs]: Can we reuse `homework` variable from line 193?

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+            //Assert
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task GetHomeworkByLessonIdAsync_NotExistingLessonId_ShouldReturnEmptyListOfHomework()
+        {
+            //Arrange
+            long lessonId = 1;
```
Comment (gingerbread-monster) [inline CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs]: Perhaps this variable should be named `nonExistentLessonId`
Please update everywhere else too.

**Review by TetianaDomanska-dev — APPROVED** [review]

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs (hunk 1 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs]: Done

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs (hunk 3 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs:185]: Done

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs (hunk 4 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs]: Done

File: CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs (hunk 5 above)
Comment (TetianaDomanska-dev) [reply CharlieBackend.Api.UnitTes/HomeworkServiceTests.cs]: Done

**Review by Serhii-Dudar — APPROVED** [review]

**Pull request merged**